doc/guide.rst

```diff
@@ -201,29 +201,40 @@ What this does is tell pyOptSparse that the key ``obj_name`` in the function ret
 For optimizers that can do multi-objective optimization (e.g. NSGA2), multiple objectives can be added.
 Optimizers that can only handle one objective enforce that only a single objective is added to the optimization description.
 
+Specifying Derivatives
+++++++++++++++++++++++
+Approximating Derivatives
+~~~~~~~~~~~~~~~~~~~~~~~~~
+pyOptSparse can automatically compute derivatives of the objective and constraint functions using finite differences or the complex-step method.
+This is done by simply passing a string to the ``sens=`` argument when calling an optimizer.
+See the possible values :ref:`here <gradient>`.
+In the simplest case, using ``sens="FD"`` will be enough to run an optimization using forward differences with a default step size.
+
 Analytic Derivatives
-++++++++++++++++++++
+~~~~~~~~~~~~~~~~~~~~
+If analytic derivatives are available, users can compute them within a user-defined function.
+This function accepts as inputs a dictionary containing design variable values as well as another dictionary containing objective and constraint values.
+It returns a nested dictionary containing the gradients of the objective and constraint values with respect to those design variables at the current design point.
+Specifically, the first-layer keys should be associated with objective and constraint names while the second-layer keys correspond to design variables.
+The dictionary values are the computed analytic derivatives, either in the form of lists or NumPy arrays whose column number equals the amount of objectives (or constraints) and row number equals the amount of design variables.
```

Comment:
I don't think 'amount' is the right word. It should just be 'number'. 
Also, how do you specify just the non-zero terms if the matrix has to be the full size?

Reply:
It's a bit hard to explain, but it's similar to the discussion above on constraints -- you can skip sub-blocks which are nonzero altogether, but within each sub-block (i.e. each `funcsSens[output_key][input_key]`) the array shapes must match. Does that make sense? How can we explain this better?

Reply:
I see what you mean about the sub-blocks. Would referring to DV groups be helpful or more confusing?

Reply:
> I don't think 'amount' is the right word. It should just be 'number'.

Yes, fixed.

Reply:
> I see what you mean about the sub-blocks. Would referring to DV groups be helpful or more confusing?

Oh I like that, I will update.

Reply:
Actually, what I said is incorrect, I believe the sub-blocks themselves can also be sparse by using a pyOptSparse or SciPy sparse array. I do think this sparsity stuff should probably be refactored into the advanced section though.

Reply:
Okay, that makes sense.